daemon/container_operations.go

```diff
@@ -594,7 +596,7 @@ func (daemon *Daemon) allocateNetwork(container *container.Container) error {
 			}
 			updateSandboxNetworkSettings(container, sb)
 			defer func() {
```

Comment:
The purpose of `defer` here is to handle the err in following `container.WriteHostConfig()`. You are right @thaJeztah .

Reply:
Yes, it's very easy to overlook the problem. My eye just fell on it, and I thought: "that doesn't look right!"